pyomo/solvers/tests/checks/test_cplex.py

```diff
@@ -54,5 +59,198 @@ def test_validate_file_name(self):
         with self.assertRaisesRegexp(ValueError, msg):
             _validate_file_name(_128, fname, 'xxx')
 
+
+class CPLEXShellWritePrioritiesFile(unittest.TestCase):
+    def setUp(self):
+        from pyomo.solvers.plugins.converter.model import PyomoMIPConverter  # register the `ProblemConverterFactory`
+        from pyomo.repn.plugins.cpxlp import ProblemWriter_cpxlp  # register the `WriterFactory`
```

Comment:
Instead if explicitly importing the plugins here, I think it would be preferable to just
```python
import pyomo.environ
```
at the top of the module (which will ensure that standard plugins are registered.